src/ProtobufProtocolSupport.chpl

```diff
@@ -492,6 +492,109 @@ module ProtobufProtocolSupport {
       return tmpObj;
     }
 
+    proc mapAppend(val, fieldNumber: int, param protoKeyType: string,
+      param protoValueType: string, ch:writingChannel) throws {
+      for (key, value) in val.items() {
+        tagAppend(fieldNumber, lengthDelimited, ch);
+        var initialOffset = ch.offset();
+        ch.mark();
+        protoFieldAppendHelper(key, 1, protoKeyType, ch);
+        protoFieldAppendHelper(value, 2, protoValueType, ch);
+        var currentOffset = ch.offset();
+        ch.revert();
+        unsignedVarintAppend((currentOffset-initialOffset):uint, ch);
+        protoFieldAppendHelper(key, 1, protoKeyType, ch);
+        protoFieldAppendHelper(value, 2, protoValueType, ch);
+      }
+    }
+
+    proc mapConsume(ch:readingChannel, ref mapField, param protoKeyType:string,
+      param protoValueType:string, type keyType, type valueType) throws {
+      var (payloadLength, _) = unsignedVarintConsume(ch);
+
+      tagConsume(ch);
+      var key = protoFieldConsumeHelper(ch, protoKeyType, keyType);
+      tagConsume(ch);
+      var value = protoFieldConsumeHelper(ch, protoValueType, valueType);
+      mapField[key] = value;
+    }
+
+    proc protoFieldAppendHelper(val, fieldNumber:int, param protoFieldType, ch:writingChannel) throws {
```

Comment:
@mppf please see this. If this works for us instead of using first class functions.

Reply:
I like it, anyway :)